tests/cephadm/test_package_validation.py

```diff
@@ -0,0 +1,156 @@
+    """
+    try:
+        # Install client packages on client node
+        for pkg, _ in client_pkgs.items():
+            if not Rpm(node).query(pkg):
+                Package(node).install(pkg, nogpgcheck=True)
+
+        # Check expected package version on client
+        for pkg, ver in client_pkgs.items():
+            if Package(node).compare(pkg, ver) == -1:
+                raise PackageError(f"Expected version for {pkg} is {ver}")
+    finally:
+        # Remove installed packages from client
+        for pkg, _ in client_pkgs.items():
+            Package(node).remove(pkg)
+
+
+def check_default_packages(node, required_pkgs):
+    """Validate the required packages
+    Args:
+        node (ceph): ceph node objects
+        required_pkgs (dict): required packages and version
+    """
+    # Check expected package version
+    for pkg, ver in required_pkgs.items():
+        if Package(node).compare(pkg, ver) == -1:
+            raise PackageError(f"Expected version for {pkg} is {ver}")
+
+
+def check_dependent_packages(node, dependent_pkgs):
+    """Validate the dependent packages
+    Args:
+        node (ceph): ceph node objects
+        dependent_pkgs (dict): dependent packages and version
+    """
+    # Check expected package version
+    for pkg, ver in dependent_pkgs.items():
+        if Package(node).compare(pkg, ver) == -1:
+            raise PackageError(f"Expected version for {pkg} is {ver}")
+
+
+def check_monitoring_images(node, monitoring_images):
+    """Validate the monitoring images
+    Args:
+        node (ceph): ceph node objects
+        monitoring_images (dict): monitoring images and version
+    """
+    reg_image = "registry.redhat.io/openshift4/{}:latest"
+
+    # Pull monitoring stack images
+    for image, ver in monitoring_images.items():
+        Container(node).pull(reg_image.format(image))
+
+    # Check expected image version
+    for image, ver in monitoring_images.items():
+        if Container(node).compare(reg_image.format(image), ver) == -1:
+            raise PackageError(
+                f"Expected version for {reg_image.format(image)} is {ver}"
+            )
+
+
+def run(ceph_cluster, **kwargs):
+    """Verify package version validation
+    Args:
+        ceph_cluster(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+    # Get installer and client node
+    client = ceph_cluster.get_nodes(role="client")[0]
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+
+    # Get config details
+    config = kwargs.get("config")
+    base_url = config.get("base_url")
+    cloud_type = config.get("cloud_type")
+
+    # Get expected package versions from config
+    client_pkgs = config.get("client_packages")
+    dependent_pkgs = config.get("dependent_packages")
+    monitoring_images = config.get("monitoring_images")
+    required_pkgs = config.get("default_packages")
+    server_pkgs = config.get("server_packages")
+
+    # Check if requied packages are installed
```

Comment:
Create methos for this one and add in below `try..catch` block

Reply:
Updated changes